### ISSUE TITLE
HystrixObservableCommand.isResponseFromFallback() return true only after fallback response is completed

### ISSUE THREAD

**Issue by allenxwang** [Netflix/Hystrix#277]
It seems that at the time ```Observer.onNext(T)``` is called, where the Observer subscribes to ```Observable<T>``` returned from HystrixObservableCommand and T is from fallback, the API HystrixObservableCommand.isResponseFromFallback() still returns false. Only when a blocking call on ```Observable<T>``` is done the API returns true.

On the other hand, HystrixObservableCommand.isFailedExecution() behaves correctly and returns true when ```Observer.onNext(T)``` is invoked and T is from fallback.

**Comment by mattrjacobs** [comment]
@allenxwang I added tests for this in #450 and couldn't replicate.  I'm closing this issue as a result.  If you've got a failing unit test, please submit a new issue.
